.github/actions/setup-coral/action.yaml

```diff
@@ -0,0 +1,49 @@
+name: Coral setup and install dependencies
+description:
+  This action installs node and pnpm in the correct version. 
+  It installs all needed dependencies and handles the pnpm store for caching. 
+  Node version and pnpm version should not be changed.
+
+inputs:
+  NODE_VERSION:
+    description: Node version for CI, matches version in .npmrc
+    default: '18'
+    required: false
+  PNPM_VERSION:
+    description: PNPM version for CI, matches version range defined in package.json
+    default: '7.13.4'
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup node.js
+      uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3.6.0
+      with:
+        node-version: ${{ inputs.NODE_VERSION }}
+
+    - name: Setup pnpm
+      uses: pnpm/action-setup@c3b53f6a16e57305370b4ae5a540c2077a1d50dd # v2.2.4
+      with:
+        version: ${{ inputs.PNPM_VERSION }}
+        run_install: false
+
+    - name: Get pnpm store directory
+      id: pnpm-cache
+      shell: bash
+      run: |
+        echo "STORE_PATH=$(pnpm store path)" >> $GITHUB_OUTPUT
+
+    - name: Setup pnpm cache
+    # @TODO set sha
```

Comment:
What is this TODO referring to? Which sha should we set?

Reply:
Forgot to remove the comment, good catch!

To define a version of an action, we either can do `uses: pnpm/action-setup@v2`, that will always use the latest 2 version. But for security reason it's recommended to refer to a specific commit hash for a version. That way, if someone manages to sneaks in malicious code in a new release of version 2, we're safe!

Reply:
That's why I updated this already to use:

`uses: pnpm/action-setup@c3b53f6a16e57305370b4ae5a540c2077a1d50dd # v2.2.4`

Forgot to remove the comment.